lib/util/cache/repository/impl/repository-cache-impl.ts

```diff
@@ -1,43 +1,36 @@
 import { promisify } from 'util';
 import zlib from 'zlib';
 import hasha from 'hasha';
-import upath from 'upath';
-import { GlobalConfig } from '../../../../config/global';
+import type { RepositoryCacheType } from '../../../../config/types';
 import { logger } from '../../../../logger';
-import { outputCacheFile, readCacheFile } from '../../../fs';
 import {
   CACHE_REVISION,
   isValidRev10,
   isValidRev11,
   isValidRev12,
 } from '../common';
-import type { RepoCacheRecord } from '../types';
+import type { CacheClient } from '../types';
 import { RepoCacheBase } from './base';
+import { CacheClientFactory } from './client/cache-client-factory';
 
 const compress = promisify(zlib.brotliCompress);
 const decompress = promisify(zlib.brotliDecompress);
 
-export class LocalRepoCache extends RepoCacheBase {
+export class RepositoryCacheImpl extends RepoCacheBase {
```

Comment:
i think the base class should now merged into this.

previously it was planned that all implementations extend from base and don't introduce another layer of classes.

can you refactor to only have this:
- one factory to create implementation
- one base class
- one local implementation
- one null implementation (for disabled cache)
- maybe an interface like below

Reply:
* one factory - done
* one base class - `RepositoryCacheImpl` now implements `RepoCache` instead of extending `RepoCacheBase`
* one local implementation - done
* one null implementation - turned base to this
* maybe an interface like below - we have `RepoCache` & `CacheClient` did you mean anything else?